managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/OpReadEntry.java

```diff
@@ -58,6 +57,11 @@ public static OpReadEntry create(ManagedCursorImpl cursor, PositionImpl readPosi
         }
         op.maxPosition = maxPosition;
         op.ctx = ctx;
+        PositionImpl position = cursor.ledger.startReadOperationOnLedger(readPositionRef, op);
+        if (position == null) {
+            return null;
```

Comment:
We can recycle this OpReadEntry if we return null.

Reply:
My point is :
- `readEntriesComplete` or `readEntriesFailed` must be called in any case, so read event would always receive response.
- `OpReadEntry` should be recyle in `readEntriesComplete` or `readEntriesFailed`.

Reply:
+1

Reply:
It might be better to skip recycling of failed entries. That was something that was recently removed for OpAddEntry instances in #12993.